src/sinks/gcs_common/config.rs

```diff
@@ -155,6 +155,7 @@ impl RetryLogic for GcsRetryLogic {
 
         match status {
             StatusCode::UNAUTHORIZED => RetryAction::Retry("unauthorized".into()),
+            StatusCode::REQUEST_TIMEOUT => RetryAction::Retry("request timeout".into()),
```

Comment:
As a follow up, can we also update the retry policy docs for the affected GC sinks?

Reply:
@pront I've raised a PR for it here: https://github.com/vectordotdev/vector/pull/21458